### PR TITLE
Fix mypy 1.1.1 errors

### LINKED ISSUE #658: fix mypy 1.1.1 fails
```
Run mypy --config-file .github/mypy/mypy.ini floss/ scripts/ tests/
floss/strings.py:55: error: Too many positional arguments for "StaticString"  [misc]
floss/strings.py:82: error: Too many positional arguments for "StaticString"  [misc]
floss/utils.py:234: error: Too many positional arguments for "StaticString"  [misc]
floss/stackstrings.py:53: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
floss/stackstrings.py:181: error: Too many positional arguments for "StackString"  [misc]
floss/tightstrings.py:112: error: Too many positional arguments for "TightString"  [misc]
floss/string_decoder.py:165: error: Too many positional arguments for "DecodedString"  [misc]
Found 6 errors in 5 files (checked 36 source files)
```

ref: #656

### PR DESCRIPTION
**Pull request by Aayush-Goel-04**
Added keywords to resolve positional arguments error.

Closes #658

### CODE REVIEW

**Comment by williballenthin** [comment]
CI build failure on mypy run. please use the following to invoke mypy as we do in CI:

```console
mypy --config-file .github/mypy/mypy.ini --check-untyped-defs floss/ scripts/ tests/
```

**Comment by Aayush-Goel-04** [comment]
> CI build failure on mypy run. please use the following to invoke mypy as we do in CI:

Looking into it

**Comment by Aayush-Goel-04** [comment]
@williballenthin 
The errors due to mypy mentined below have been resolved. Please look at the changes when you have a chance.
```
run: mypy --config-file .github/mypy/mypy.ini --check-untyped-defs floss/ scripts/ tests/
floss/results.py:193: error: "Type[ResultDocument]" has no attribute "__pydantic_model__"  [attr-defined]
floss/utils.py:310: error: Module has no attribute "builtins"; maybe "isbuiltin"?  [attr-defined]
floss/utils.py:3[13](https://github.com/mandiant/flare-floss/actions/runs/4886852254/jobs/8724169319?pr=720#step:7:14): error: Module has no attribute "builtins"; maybe "isbuiltin"?  [attr-defined]
floss/render/default.py:277: error: "TextIO" has no attribute "reconfigure"  [attr-defined]
floss/render/default.py:308: error: "IO[str]" has no attribute "getvalue"  [attr-defined]
floss/features/features.py:[15](https://github.com/mandiant/flare-floss/actions/runs/4886852254/jobs/8724169319?pr=720#step:7:16)6: error: Argument 1 to "float" has incompatible type "Union[Any, float, None]"; expected "Union[SupportsFloat, SupportsIndex, str, bytes, bytearray, memoryview, array[Any], mmap, _CData, PickleBuffer]"  [arg-type]
floss/main.py:266: error: Module has no attribute "TRACE"  [attr-defined]
scripts/idaplugin.py:[19](https://github.com/mandiant/flare-floss/actions/runs/4886852254/jobs/8724169319?pr=720#step:7:20)4: error: Argument 1 to "apply_stack_strings" has incompatible type "List[TightString]"; expected "List[Union[StackString, TightString]]"  [arg-type]
scripts/idaplugin.py:194: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
scripts/idaplugin.py:194: note: Consider using "Sequence" instead, which is covariant
tests/conftest.py:35: error: Incompatible types in assignment (expression has type "StackString", variable has type "DecodedString")  [assignment]
tests/conftest.py:39: error: Incompatible types in assignment (expression has type "TightString", variable has type "DecodedString")  [assignment]
tests/conftest.py:57: error: LEGACY_PATH? has no attribute "open"  [attr-defined]
```

File: floss/results.py [hunk 1]
```diff
@@ -189,7 +190,7 @@ class ResultDocument:
 
     @classmethod
     def parse_file(cls, path):
-        return cls.__pydantic_model__.parse_file(path)
+        return cls.parse_file(path)
```
Comment (williballenthin) [inline floss/results.py:193]: this results in an infinite loop:

https://github.com/mandiant/flare-floss/actions/runs/4902927902/jobs/8819760807?pr=720

one solution might be to use the mypy plugin described here: https://docs.pydantic.dev/latest/usage/dataclasses/, would you give that a shot? if it doesn't work, i have an idea for a fallback or two.

Comment (Aayush-Goel-04) [reply floss/results.py:193]: The mypy puligin  as mentioned in https://docs.pydantic.dev/latest/mypy_plugin/#enabling-the-plugin is already enabled in mypy.ini.

Comment (mr-tz) [reply floss/results.py:193]: besides this everything else looks good

Comment (williballenthin) [reply floss/results.py:193]: lets revert this change, add a `  # type: ignore` to the line, and a comment that explains that this field is guaranteed by the pydantic dataclass.

Comment (Aayush-Goel-04) [reply floss/results.py]: @williballenthin , I have made changes. Please have a look at it.

**Review by williballenthin — APPROVED** [review]
tests pass and the changes look great, we're golden. thanks @Aayush-Goel-04!

**Comment by Aayush-Goel-04** [comment]
Thanks for the Feedback! Looking forward to make more contributions.

**Pull request merged**